apps/mobile/src/components/Feed/createVirtualizedFeedEventItems.ts

```diff
@@ -370,5 +382,28 @@ export function createVirtualizedFeedEventItems({
     }
   }
 
-  return { items };
+  let postCount = 0;
+  const itemsWithSuggestedProfileRow = [];
+  let insertedRow = false;
+
+  for (const item of items) {
+    if (!insertedRow && item.kind === 'post-item-header') {
+      postCount++;
+      if (postCount === SUGGESTED_PROFILE_ROW_IDX) {
+        itemsWithSuggestedProfileRow.push({
+          kind: 'suggested-profile-row',
+          event: null,
+          post: null,
+          queryRef: query,
+          key: 'suggested-profile-row',
+          eventId: 'suggested-profile-row',
+          itemType: null,
+        } as FeedListItemType);
+        insertedRow = true;
+      }
+    }
+    itemsWithSuggestedProfileRow.push(item);
+  }
```

Comment:
1. could we add an explainer / comment block for what's going on here?
2. do we need to mark `insertedRow` as `true`? i assume `postCount === SUGGESTED_PROFILE_ROW_IDX` would only be true a single time anyway?

overall i'm wondering if there could be a simpler way to do this without the loop, like:
```ts
const itemsWithSuggestedProfileRow = [
  ...items.slice(0, 8),
  { kind: 'suggested-profile-row', /* rest of the object */ }
  ...items.slice(8)
]
```

Reply:
yo @Robinnnnn, you're right we don't need the `insertedRow` variable, removed that now. 

Regarding using the splice, I think we can't use the hardcoded value for the index because the items of the list are `feed-post-header`, `feed-post-socialize-section`, etc (basically subsections of a single post in the feed for all the posts) so we want to add the row only when it is before a `feed-post-header` 

Or it might end up after a post and before its mint button for example.

<img width="814" alt="Screenshot 2024-04-12 at 00 28 45" src="https://github.com/gallery-so/gallery/assets/49758803/757bc124-7b15-4e94-8ec7-c1005a9298d0">